src/VecSim/algorithms/hnsw/hnsw_single.h

```diff
@@ -98,25 +97,35 @@ void HNSWIndex_Single<DataType, DistType>::resizeLabelLookup(size_t new_max_elem
 
 template <typename DataType, typename DistType>
 int HNSWIndex_Single<DataType, DistType>::deleteVector(const labelType label) {
-    // check that the label actually exists in the graph, and update the number of elements.
+    // Check that the label actually exists in the graph, and update the number of elements.
     if (label_lookup_.find(label) == label_lookup_.end()) {
-        return false;
+        return 0;
     }
     idType element_internal_id = label_lookup_[label];
     label_lookup_.erase(label);
-    return this->removeVector(element_internal_id);
+    this->removeVector(element_internal_id);
+    return 1;
 }
 
 template <typename DataType, typename DistType>
-int HNSWIndex_Single<DataType, DistType>::addVector(const void *vector_data,
-                                                    const labelType label) {
+int HNSWIndex_Single<DataType, DistType>::addVector(const void *vector_data, const labelType label,
+                                                    bool override_allowed) {
 
-    // Checking if an element with the given label already exists. if so, remove it.
+    // Checking if an element with the given label already exists.
+    size_t index_size_before = this->indexSize();
     if (label_lookup_.find(label) != label_lookup_.end()) {
-        deleteVector(label);
+        if (override_allowed) {
+            // Remove the vector in place if override allowed (in non-async scenario)
+            deleteVector(label);
+        } else {
+            // If override is not allowed, we don't do anything.
+            return -1;
+        }
     }
 
-    return this->appendVector(vector_data, label);
+    this->appendVector(vector_data, label);
+    // Return the delta in the index size due to the insertion.
+    return this->indexSize() - index_size_before;
```

Comment:
why not 1?

Reply:
Because in case of override, we should return 0